### ISSUE TITLE
Members Streams

### ISSUE THREAD

**Issue by Flat** [HizkiFW/youtube-stream-status#1]
I've been trying to work out how to use the same hearbeat check to see if a member's stream is live by passing a netscape cookies file, however youtube doesn't seem to think I'm logged in. Any ideas on what fields are necessary for this? YouTube will return json status such as `'playabilityStatus': {'status': 'LIVE_STREAM_OFFLINE'}` but not the `liveSteamabilityRenderer`

**Comment by hizkifw** [comment]
I don't have a membership to anyone so I can't test, but you should be able to pass cookies to `urllib.request` using [this method](https://stackoverflow.com/a/51682831). You'd need to insert each cookie using [`CookieJar.set_cookie`](https://docs.python.org/3/library/http.cookiejar.html#http.cookiejar.CookieJar.set_cookie).

I might update this sometime in the future to use [`requests`](https://3.python-requests.org/) instead to make it easier to use cookies.

**Comment by Flat** [comment]
I did switch the code over to requests and tried using a session. Maybe I need to actually hit the login page with the cookies. I'll post the current code changes I've made to a gist if you're interested, in about 9 hours when I can. Suppose I should examine closer how youtube-dl does this. Thanks!

**Comment by Flat** [comment]
Here is the gist with the changes I made as well as part of the JSON response including noticeably `logged_in: 0` and `loggedOut True` (This is an offline example as there are no pending member streams that I can find, but the result is the same regardless)
https://gist.github.com/Flat/640b510f1b239683504554ff8f82486f
```
{'service': 'GUIDED_HELP', 'params': [{'key': 'logged_in', 'value': '0'}]}, {'service': 'ECATCHER', 'params': [{'key': 'client.version', 'value': '2.20210323'}, {'key': 'client.name', 'value': 'WEB'}]}], 'mainAppWebResponseContext': {'loggedOut': True}, 'webResponseContextExtensionData': {'hasDecorated': True}}, 'playabilityStatus': {'status': 'LIVE_STREAM_OFFLINE'}}
```

**Comment by hizkifw** [comment]
I tried passing cookies directly using the `Cookie` header and got a similar response when tested on a private scheduled stream.

    'playabilityStatus': {'status': 'UNPLAYABLE', 'reason': 'This video is unavailable.'}

I'm not sure what sort of authentication the heartbeat endpoint needs, I'll have to take a deeper look into this. Alternatively maybe you can use the [push notification API](https://developers.google.com/youtube/v3/guides/push_notifications).

**Comment by hizkifw** [comment]
Fixed in 3c3fa17eda49fa54c442615a7b7a44ac9b04b93f
